charts/newrelic-infrastructure-v3/templates/ksm/configmap.yaml

```diff
@@ -1,4 +1,5 @@
-{{- if .Values.ksm.enabled -}}
+{{- $ksmEnabled := ((.Values.ksm.enabled | default (not .Values.ksm.disableKubeStateMetrics))) -}}
```

Comment:
Even if redundant, I'd suggest to wrap this into a helper so we can later identify it as such. Otherwise we might forget that `.Values.ksm.disableKubeStateMetrics` was in fact legacy :)

Reply:
Moreover, this line might not behave as the user expect. On a new install, `.Values.ksm.enabled` will default to `true`, effectively overriding an old `.Values.ksm.disableKubeStateMetrics: false`. I think we might need to wrap the logic in a helper to process priority as it should.

Reply:
true, I'll take care of that